dev/tools/bin/find_commit.dart

```diff
@@ -43,41 +29,37 @@ String findCommit({
   required String secondaryRepoDirectory,
   required String secondaryBranch,
 }) {
-  final Commit anchor;
+  final DateTime anchor;
   if (primaryBranch == primaryTrunk) {
-    log('on $primaryTrunk, using last commit as anchor');
-    anchor = Commit.parse(git(primaryRepoDirectory, <String>['log', Commit.formatArgument, '--max-count=1', primaryBranch, '--']));
+    log('on $primaryTrunk, using last commit time');
+    anchor = _parseTimestamp(git(primaryRepoDirectory, <String>['log', _commitTimestampFormat, '--max-count=1', primaryBranch, '--']));
   } else {
-    final List<Commit> branchCommits = Commit.parseList(git(primaryRepoDirectory, <String>['log', Commit.formatArgument, primaryBranch, '--']));
-    final List<Commit> trunkCommits = Commit.parseList(git(primaryRepoDirectory, <String>['log', Commit.formatArgument, primaryTrunk, '--']));
-    if (branchCommits.isEmpty || trunkCommits.isEmpty || branchCommits.first.hash != trunkCommits.first.hash) {
+    final String mergeBase = git(primaryRepoDirectory, <String>['merge-base', primaryBranch, primaryTrunk], allowFailure: true).trim();
+    if (mergeBase.isEmpty) {
       throw StateError('Branch $primaryBranch does not seem to have a common history with trunk $primaryTrunk.');
     }
-    if (branchCommits.last.hash == trunkCommits.last.hash) {
-      log('$primaryBranch is even with $primaryTrunk, using last commit as anchor');
-      anchor = trunkCommits.last;
+    anchor = _parseTimestamp(git(primaryRepoDirectory, <String>['log', _commitTimestampFormat, '--max-count=1', mergeBase, '--']));
+    final int missingTrunkCommits = _countLines(git(primaryRepoDirectory, <String>['rev-list', primaryTrunk, '^$primaryBranch', '--']));
```

Comment:
we could skip these processes by surrounding this with a `debugLogging` check now, I may as well add that optimization.